Index: department-service/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- department-service/pom.xml	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ department-service/pom.xml	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -56,12 +56,20 @@
 			<version>2.8.0</version>
 		</dependency>
 		<dependency>
-		    <groupId>org.springframework.cloud</groupId>
-		    <artifactId>spring-cloud-starter-sleuth</artifactId>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-starter-bus-amqp</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.baeldung.spring.cloud</groupId>
+			<artifactId>spring-cloud-hystrix-rest-producer</artifactId>
 		</dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-hystrix</artifactId>
+        </dependency>
 		<dependency>
 			<groupId>org.springframework.cloud</groupId>
-			<artifactId>spring-cloud-starter-bus-amqp</artifactId>
+			<artifactId>spring-cloud-starter-netflix-hystrix</artifactId>
 		</dependency>
 	</dependencies>
 
Index: department-service/src/main/java/pl/piomin/services/department/DepartmentApplication.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- department-service/src/main/java/pl/piomin/services/department/DepartmentApplication.java	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ department-service/src/main/java/pl/piomin/services/department/DepartmentApplication.java	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -2,6 +2,7 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.client.circuitbreaker.EnableCircuitBreaker;
 import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
 import org.springframework.cloud.openfeign.EnableFeignClients;
 import org.springframework.context.annotation.Bean;
@@ -18,6 +19,7 @@
 @SpringBootApplication
 @EnableDiscoveryClient
 @EnableFeignClients
+@EnableCircuitBreaker
 @EnableSwagger2
 public class DepartmentApplication {
 
Index: department-service/src/main/java/pl/piomin/services/department/client/EmployeeClient.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- department-service/src/main/java/pl/piomin/services/department/client/EmployeeClient.java	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ department-service/src/main/java/pl/piomin/services/department/client/EmployeeClient.java	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -8,7 +8,7 @@
 
 import pl.piomin.services.department.model.Employee;
 
-@FeignClient(name = "employee-service")
+@FeignClient(name = "employee-service", fallback = EmployeeClientFallback.class)
 public interface EmployeeClient {
 
 	@GetMapping("/department/{departmentId}")
Index: department-service/src/main/java/pl/piomin/services/department/client/EmployeeClientFallback.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- department-service/src/main/java/pl/piomin/services/department/client/EmployeeClientFallback.java	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
+++ department-service/src/main/java/pl/piomin/services/department/client/EmployeeClientFallback.java	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -0,0 +1,15 @@
+package pl.piomin.services.department.client;
+
+import org.springframework.stereotype.Service;
+import pl.piomin.services.department.model.Employee;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class EmployeeClientFallback implements EmployeeClient{
+    @Override
+    public List<Employee> findByDepartment(Long departmentId) {
+        return new ArrayList<>();
+    }
+}
Index: department-service/src/main/resources/bootstrap.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- department-service/src/main/resources/bootstrap.yml	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ department-service/src/main/resources/bootstrap.yml	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -9,4 +9,7 @@
         output:
           destination: foodOrders
       default:
-        contentType: application/json
\ No newline at end of file
+        contentType: application/json
+feign:
+  hystrix:
+    enabled: true
\ No newline at end of file
Index: employee-service/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- employee-service/pom.xml	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ employee-service/pom.xml	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -50,10 +50,6 @@
         </dependency>
         <dependency>
             <groupId>org.springframework.cloud</groupId>
-            <artifactId>spring-cloud-starter-sleuth</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.springframework.cloud</groupId>
             <artifactId>spring-cloud-starter-bus-amqp</artifactId>
         </dependency>
     </dependencies>
Index: employee-service/src/main/java/pl/piomin/services/employee/controller/EmployeeController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- employee-service/src/main/java/pl/piomin/services/employee/controller/EmployeeController.java	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ employee-service/src/main/java/pl/piomin/services/employee/controller/EmployeeController.java	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -42,6 +42,11 @@
 	
 	@GetMapping("/department/{departmentId}")
 	public List<Employee> findByDepartment(@PathVariable("departmentId") Long departmentId) {
+		try {
+			Thread.sleep(1000*1000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
 		LOGGER.info("Employee find: departmentId={}", departmentId);
 		return repository.findByDepartment(departmentId);
 	}
Index: employee-service/src/main/resources/bootstrap.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- employee-service/src/main/resources/bootstrap.yml	(revision 47ae3ec8c82f648314fff6bdc1b7d6bc50ab8867)
+++ employee-service/src/main/resources/bootstrap.yml	(revision 03b7b5819316da3bf8bc3c861fc2dfe5a0a1c11d)
@@ -2,7 +2,7 @@
   application:
     name: employee-service
   profiles:
-    active: instance2
+    active: instance
   cloud:
     config:
       uri: http://localhost:8088
@@ -16,3 +16,5 @@
     port: 5672
     username: guest
     password: guest
+
+feign
